cli/src/main/java/com/devonfw/tools/ide/tool/intellij/IntellijPluginManager.java

```diff
@@ -0,0 +1,140 @@
+package com.devonfw.tools.ide.tool.intellij;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import com.devonfw.tools.ide.context.IdeContext;
+import com.devonfw.tools.ide.io.FileAccess;
+import com.devonfw.tools.ide.step.Step;
+import com.devonfw.tools.ide.tool.ide.PluginDescriptor;
+
+/**
+ * Manager class to install plugins for the {@link Intellij commandlet}.
+ */
+public class IntellijPluginManager {
+
+  private static final String BUILD_FILE = "build.txt";
+  private final IdeContext context;
+  private final Intellij commandlet;
+
+  /**
+   * The constructor
+   *
+   * @param context the {@link IdeContext}.
+   * @param commandlet the {@link Intellij commandlet}.
+   */
+  public IntellijPluginManager(IdeContext context, Intellij commandlet) {
+    this.context = context;
+    this.commandlet = commandlet;
+  }
+
+  /**
+   * Installs a plugin.
+   *
+   * @param plugin the {@link PluginDescriptor}#.
+   */
+  public void installPlugin(PluginDescriptor plugin) {
+    String pluginId = plugin.getId();
+    String downloadUrl = plugin.getUrl();
+
+    Path tmpDir = null;
+
+    Step step = this.context.newStep("Install plugin: " + pluginId);
+    try {
+      String buildVersion = readBuildVersion();
+
+      Path installationPath = commandlet.getPluginsInstallationPath();
+      ensureInstallationPathExists(installationPath);
+
+      if (downloadUrl == null || downloadUrl.isEmpty()) {
+        downloadUrl = String.format("https://plugins.jetbrains.com/pluginManager?action=download&id=%s&build=%s", pluginId, buildVersion);
```

Comment:
Maybe move this link to a constant too?

Reply:
I think I would leave it like that to make it easier for the developer to see where the values or the arguments are gonna be replaced in the string replacing `%s`